hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/BaseFileOnlyRelation.scala

```diff
@@ -101,4 +102,17 @@ class BaseFileOnlyRelation(sqlContext: SQLContext,
 
     sparkAdapter.getFilePartitions(sparkSession, fileSplits, maxSplitBytes).map(HoodieBaseFileSplit.apply)
   }
+
+  def toHadoopFsRelation: HadoopFsRelation = {
+    HadoopFsRelation(
+      fileIndex,
+      fileIndex.partitionSchema,
+      fileIndex.dataSchema,
+      bucketSpec = None,
+      fileFormat = metaClient.getTableConfig.getBaseFileFormat match {
+        case HoodieFileFormat.PARQUET => new ParquetFileFormat
+        case HoodieFileFormat.ORC => new OrcFileFormat
+      },
+      optParams)(sparkSession)
```

Comment:
maybe we need HoodieFileIndex or  HoodieROTablePathFilter to filter old parquet file.
how about fallabck to the implement of 0.10.0     https://github.com/apache/hudi/blob/cc3896be2a023f8819883d745178503286ac2ab1/hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/DefaultSource.scala#L182